collection-scripts/gather_services_status

```diff
@@ -3,37 +3,71 @@
 
 alias os="/usr/bin/oc -n openstack rsh openstackclient openstack "
 
+# For each service passed an input, if the associated entry exists,
+# we can call the related function that processes specific service
+# commands
 get_status() {
     service="$1"
+    echo "Gather ctlplane service info: $service"
     case "${service}" in
-    "openstack") get_openstack_status
-                 ;;
-    "manila") get_manila_status
-              ;;
-    *) exit 0
+    "openstack")
+        get_openstack_status
+        ;;
+    "manila")
+        get_manila_status
+        ;;
+    "neutron")
+        get_neutron_status
+        ;;
+    *) ;;
     esac
 }
 
+# Generic OpenStack cltplane gathering -
 get_openstack_status() {
-        mkdir -p "$BASE_COLLECTION_PATH"/ctlplane
-        ${BASH_ALIASES[os]} endpoint list > "$BASE_COLLECTION_PATH"/ctlplane/endpoints
-        ${BASH_ALIASES[os]} service list > "$BASE_COLLECTION_PATH"/ctlplane/services
-        ${BASH_ALIASES[os]} compute service list > "$BASE_COLLECTION_PATH"/ctlplane/compute_service_list
-        ${BASH_ALIASES[os]} network agent list > "$BASE_COLLECTION_PATH"/ctlplane/network_agent_list
+    mkdir -p "$BASE_COLLECTION_PATH"/ctlplane
+    ${BASH_ALIASES[os]} endpoint list > "$BASE_COLLECTION_PATH"/ctlplane/endpoints
+    ${BASH_ALIASES[os]} service list > "$BASE_COLLECTION_PATH"/ctlplane/services
+    ${BASH_ALIASES[os]} compute service list > "$BASE_COLLECTION_PATH"/ctlplane/compute_service_list
+    ${BASH_ALIASES[os]} network agent list > "$BASE_COLLECTION_PATH"/ctlplane/network_agent_list
 }
 
+# Manila service gathering -
 get_manila_status() {
-    ${BASH_ALIASES[os]} share service list > "$BASE_COLLECTION_PATH"/manila/service_list
-    ${BASH_ALIASES[os]} share type list > "$BASE_COLLECTION_PATH"/manila/share_types
-    ${BASH_ALIASES[os]} share pool list --detail > "$BASE_COLLECTION_PATH"/manila/pool_list
+    local MANILA_PATH="$BASE_COLLECTION_PATH/ctlplane/manila"
+    mkdir -p "$MANILA_PATH"
+    ${BASH_ALIASES[os]} share service list > "$MANILA_PATH"/service_list
+    ${BASH_ALIASES[os]} share type list > "$MANILA_PATH"/share_types
+    ${BASH_ALIASES[os]} share pool list --detail > "$MANILA_PATH"/pool_list
 }
 
-for svc in "${SERVICES[@]}"; do
-    get_status "$svc"
+# Neutron service gathering -
+get_neutron_status() {
+    local NEUTRON_PATH="$BASE_COLLECTION_PATH/ctlplane/neutron"
+    mkdir -p "$NEUTRON_PATH"
+    ${BASH_ALIASES[os]} subnet list > "$NEUTRON_PATH"/subnet_list
+    ${BASH_ALIASES[os]} port list > "$NEUTRON_PATH"/port_list
+    ${BASH_ALIASES[os]} router list > "$NEUTRON_PATH"/router_list
+    ${BASH_ALIASES[os]} network agent list > "$NEUTRON_PATH"/agent_list
+    ${BASH_ALIASES[os]} network list > "$NEUTRON_PATH"/network_list
+    ${BASH_ALIASES[os]} extension list > "$NEUTRON_PATH"/extension_list
+    ${BASH_ALIASES[os]} floating ip list > "$NEUTRON_PATH"/floating_ip_list
+    ${BASH_ALIASES[os]} security group lis > "$NEUTRON_PATH"/security_group_list
```

Comment:
list

Reply:
There are probably more resources that we may want to collect information about (SG rules, trunks) but this is a great start and we can adjust later.

Reply:
Ack and thanks for the quick review. I fixed the typo (lis vs list) and I agree we can add later whatever we need to gather. The nice thing here is having a clear (and simple) interface to improve the way we gather info about neutron in the existing ctlplane.